compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1690,6 +1690,16 @@ impl<'hir> LoweringContext<'_, 'hir> {
         let yielded =
             opt_expr.as_ref().map(|x| self.lower_expr(x)).unwrap_or_else(|| self.expr_unit(span));
 
+        if !self.tcx.features().yield_expr() {
```

Comment:
Maybe make this 
```suggestion
        if !self.tcx.features().yield_expr() && !self.tcx.features().coroutines() {
```

just to reduce the churn and make using coroutines simpler